### DISCUSSION TITLE
Reverse stop working (php-curl)

### DISCUSSION THREAD

**Discussion by alebalweb** [osm-search/Nominatim#3518] [Q&A]
Why this stop working?

`function reversegeocode($lat, $lon){

    $url = "https://nominatim.openstreetmap.org/reverse?lat=".$lat."&lon=".$lon."&zoom=10&format=json&addressdetails=1&accept-language=en";

    $ch = curl_init();
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_URL,$url);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
    curl_setopt($ch, CURLOPT_HEADER, false);
    curl_setopt($ch, CURLOPT_REFERER, $url);
    curl_setopt($ch, CURLOPT_USERAGENT, "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.86 Safari/537.36");

    $result = curl_exec($ch);

    curl_close($ch);

    return json_decode($result, true);
}`

worked for years, I missed something lately?

**Comment by mtmail** [comment]
Set a user agent which identifies your app, it can also be an email address 
https://operations.osmfoundation.org/policies/nominatim/

**Comment by alebalweb** [comment]
Dont work...

`function reversegeocode($lat, $lon){

    $url = "https://nominatim.openstreetmap.org/reverse?lat=".$lat."&lon=".$lon."&zoom=10&format=json&addressdetails=1&accept-language=en";

    $ch = curl_init();
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_URL,$url);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
    curl_setopt($ch, CURLOPT_HEADER, false);
    curl_setopt($ch, CURLOPT_REFERER, $url);
    //curl_setopt($ch, CURLOPT_USERAGENT, "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.86 Safari/537.36");
    curl_setopt($ch, CURLOPT_USERAGENT, _SITE_NAME." ReverseGeocode service");

    $result = curl_exec($ch);

    curl_close($ch);

    return json_decode($result, true);
}`

**Comment by alebalweb** [comment]
Not sure why, but looks like it work like this

`function reversegeocode($lat, $lon){

    $url = "https://nominatim.openstreetmap.org/reverse?lat=".$lat."&lon=".$lon."&zoom=10&format=json&addressdetails=1&accept-language=en";

    $ch = curl_init();
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_URL,$url);
    //curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
    //curl_setopt($ch, CURLOPT_HEADER, false);
    //curl_setopt($ch, CURLOPT_REFERER, $url);
    //curl_setopt($ch, CURLOPT_USERAGENT, "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.86 Safari/537.36");
    curl_setopt($ch, CURLOPT_USERAGENT, _SITE_NAME." ReverseGeocode service");

    $result = curl_exec($ch);

    curl_close($ch);

    return json_decode($result, true);
}

someone can explain?`

**Reply by mtmail** [reply]
The serice (nominatim.openstreetmap.org) blocks IP addresses. It's automatic. I think your IP address was un-blocked after you made the change to the user agent.
